source/extensions/network/dns_resolver/cares/dns_impl.h

```diff
@@ -45,10 +45,12 @@ class DnsResolverImplPeer;
  */
 class DnsResolverImpl : public DnsResolver, protected Logger::Loggable<Logger::Id::dns> {
 public:
+  static absl::StatusOr<absl::optional<std::string>>
+  maybeBuildResolversCsv(const std::vector<Network::Address::InstanceConstSharedPtr>& resolvers);
+
   DnsResolverImpl(
       const envoy::extensions::network::dns_resolver::cares::v3::CaresDnsResolverConfig& config,
-      Event::Dispatcher& dispatcher,
-      const std::vector<Network::Address::InstanceConstSharedPtr>& resolvers,
+      Event::Dispatcher& dispatcher, absl::optional<std::string> resolvers_csv,
```

Comment:
Just curious what's the motivation for this change?

Reply:
envoyproxy/envoy-mobile#176 :-)

but more relevant to Envoy, there's also https://github.com/envoyproxy/envoy/issues/27412